aineko/extras/connectors/rest_poller.py

```diff
@@ -0,0 +1,260 @@
+                response = self.session.get(
+                    self.rest_params.url,
+                    timeout=self.rest_params.timeout,
+                    headers=self.rest_params.headers,
+                    data=self.rest_params.data,
+                )
+                # Check if the request was successful
+                if response.status_code not in self.rest_params.success_codes:
+                    # pylint: disable=broad-exception-raised
+                    raise Exception(
+                        f"Request to url {self.rest_params.url} "
+                        "failed with status code: "
+                        f"{response.status_code}"
+                    )
+                raw_message = response.text
+            except Exception as err:  # pylint: disable=broad-except
+                # If request fails, log the error and sleep
+                self.log(
+                    "Request failed. "
+                    f"Sleeping for {self.rest_params.poll_interval} seconds. "
+                    f"Error: {err}",
+                    level="error",
+                )
+                time.sleep(self.rest_params.poll_interval)
+                self.retry_count += 1
+                # Reset the session
+                self.log(
+                    "Creating new session to REST endpoint "
+                    f"{self.rest_params.url}."
+                )
+                self.session = requests.Session()
+                return
+
+            try:
+                # Parse the message and emit to producers
+                message = json.loads(raw_message)
+                if self.rest_params.metadata is not None:
+                    message = {
+                        "metadata": self.rest_params.metadata,
+                        "data": message,
+                    }
+                self.producers[self.output_dataset].produce(message)
+                self.retry_count = 0
+            except json.decoder.JSONDecodeError as err:
+                if self.retry_count < self.rest_params.max_retries:
+                    self.retry_count += 1
+                    self.log(
+                        f"Failed to parse message: {raw_message}. "
+                        f"The following error occurred: {err} "
+                        f"Will retry in {self.rest_params.retry_sleep} "
+                        "seconds...",
+                        level="error",
+                    )
+                    time.sleep(self.rest_params.retry_sleep)
+                else:
+                    raise Exception(  # pylint: disable=broad-exception-raised
+                        "Retry count exceeded max retries "
+                        f"({self.rest_params.max_retries}). "
+                        f"Failed to parse message: {raw_message}. "
+                        f"The following error occurred: {err}"
+                    ) from err
+        else:
+            # If it is not time to poll, sleep
+            time.sleep(
```

Comment:
I'm pretty sure this number could be negative and the `time` library would raise a `ValueError`.

Example:
`self.rest_params.poll_interval` = 0.1
`time.time()` = 1706198047
`self.last_poll_time` = 1706198042 (implies the API call took roughly 2 seconds)

Reply:
The result of this math would be -4.9

Reply:
By construction, we reach this logic if and only if `time.time()` - `self.last_poll_time` < `self.rest_params.poll_interval` so thus the scenario you describe would never occur. If the API takes 2 seconds to respond, then on the next iteration through the loop, the code would flow into the polling logic.